pkg/commands/config.go

```diff
@@ -5,15 +5,27 @@ import (
 	"os"
 
 	"github.com/spf13/cobra"
-	"github.com/spf13/pflag"
 	"github.com/spf13/viper"
 
 	"github.com/golangci/golangci-lint/pkg/config"
 	"github.com/golangci/golangci-lint/pkg/exitcodes"
 	"github.com/golangci/golangci-lint/pkg/fsutils"
+	"github.com/golangci/golangci-lint/pkg/logutils"
 )
 
-func (e *Executor) initConfig() {
+type configCommand struct {
+	viper *viper.Viper
```

Comment:
some offtop: do we really need the `viper` dependency?

why not just declare the strict ways to use golangci-lint:

1. flags for command itself settings
2. config for linter specific settings

without intersection

Reply:
> some offtop: do we really need the viper dependency?

I think at some point I will remove viper.

> why not just declare the strict ways to use golangci-lint:
>
>    flags for command itself settings
>    config for linter specific settings
>
> without intersection

It's useful to be be the mix both on linters (ex: `-E`, `-D`)